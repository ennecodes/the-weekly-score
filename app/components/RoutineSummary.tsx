import { Link } from "@remix-run/react";

import { RoutineCard } from "./RoutineCard";
import { RoutineDTO } from "./types";

export const RoutineSummary = ({ routines }: { routines: RoutineDTO[] }) => {
  return (
    <>
      <div className="grid grid-cols-1 gap-4 md:grid-cols-2 lg:grid-cols-3">
        {routines.map((routine, index) => (
          <RoutineCard key={index} routine={routine} />
        ))}
      </div>
      <Link
        to={{
          pathname: "/addRoutine",
        }}
      >
        <button className="rounded bg-pink-500 px-4 py-2 font-bold text-white hover:bg-pink-700">
          Add Routine
        </button>
      </Link>
    </>
  );
};
