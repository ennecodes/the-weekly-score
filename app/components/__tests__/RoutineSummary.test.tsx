import { render, screen} from "@testing-library/react";

import { RoutineSummary } from "../RoutineSummary";

describe("RoutineSummary", () => {
    it("renders correctly", () => {
        const { container } = render(<RoutineSummary routines={[]} />);
        expect(container).toMatchSnapshot();
    });
    it("displays the routine cards", () => {
        const testRoutines = [
            {id: "1", name: "Routine 1", tasks: []},
            {id: "2", name: "Routine 2", tasks: []},
            {id: "3", name: "Routine 3", tasks: []},
        ];
        render(<RoutineSummary routines={testRoutines} />);
        expect(screen.getByText("Routine 1")).toBeInTheDocument();
        expect(screen.getByText("Routine 2")).toBeInTheDocument();
        expect(screen.getByText("Routine 3")).toBeInTheDocument();
    });
    it("Allows the user to add a new routine", () => {
        const testRoutines = [
            {id: "1", name: "Routine 1", tasks: []},
            {id: "2", name: "Routine 2", tasks: []},
            {id: "3", name: "Routine 3", tasks: []},
        ];
        const { container } = render(<RoutineSummary routines={testRoutines} />);
        const button = screen.getByText("Add Routine");
        button.click();
        expect(container).toMatchSnapshot();
    }
});