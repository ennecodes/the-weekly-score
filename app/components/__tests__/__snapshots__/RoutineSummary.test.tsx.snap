// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RoutineSummary > renders correctly 1`] = `
<div>
  <div
    class="grid grid-cols-1 gap-4 md:grid-cols-2 lg:grid-cols-3"
  />
</div>
`;
